src/DiffEqGPU.jl

```diff
@@ -65,44 +48,41 @@ function GPUifyLoops.launch_config(::Union{typeof(gpu_kernel),
     blocks = ceil(Int,size(u,2)/t)
     (threads=t,blocks=blocks)
 end
+=#
 
-function W_kernel(jac, W, u, p, gamma, t)
+@kernel function W_kernel(jac, W, u, p, gamma, t)
+    i = @index(Global, Linear)
     len = size(u,1)
-    @loop for i in (1:size(u,2); (blockIdx().x-1) * blockDim().x + threadIdx().x)
-        _W = @inbounds @view(W[:, :, i])
-        @views @inbounds jac(_W,u[:,i],p[:,i],t)
-        @inbounds for i in eachindex(_W)
-            _W[i] = gamma*_W[i]
-        end
-        _one = one(eltype(_W))
-        @inbounds for i in 1:len
-            _W[i, i] = _W[i, i] - _one
-        end
-        nothing
+    _W = @inbounds @view(W[:, :, i])
+    @views @inbounds jac(_W,u[:,i],p[:,i],t)
+    @inbounds for i in eachindex(_W)
+        _W[i] = gamma*_W[i]
+    end
+    _one = one(eltype(_W))
+    @inbounds for i in 1:len
+        _W[i, i] = _W[i, i] - _one
     end
-    return nothing
 end
 
-function Wt_kernel(jac, W, u, p, gamma, t)
+@kernel function Wt_kernel(jac, W, u, p, gamma, t)
+    i = @index(Global, Linear)
     len = size(u,1)
-    @loop for i in (1:size(u,2); (blockIdx().x-1) * blockDim().x + threadIdx().x)
-        _W = @inbounds @view(W[:, :, i])
-        @views @inbounds jac(_W,u[:,i],p[:,i],t)
-        @inbounds for i in 1:len
-            _W[i, i] = -inv(gamma) + _W[i, i]
-        end
-        nothing
+    _W = @inbounds @view(W[:, :, i])
+    @views @inbounds jac(_W,u[:,i],p[:,i],t)
```

Comment:
do you need to use views in the kernels? I thought they would use CuDeviceArrays like GPUifyLoops?

Reply:
I'm not sure, I just kept them there by default. These kernels will have their arguments converted to `CuDeviceArray` via Adapt. If that's the case, do we need views? Is there some discussion that explains why they were originally needed?

Reply:
The view is because it's a mutating function, so I guess they need to be kept so that the reference still points back. However, I wonder if KernelAbstractions is keeping that.

Reply:
@vchuavry

Reply:
Yeah these are CuDeviceArray as in GPUifyLoops, so the views are necessary